app/controllers/idv/in_person/address_controller.rb

```diff
@@ -73,7 +99,14 @@ def confirm_in_person_state_id_step_complete
       def confirm_in_person_address_step_needed
         return if pii_from_user && pii_from_user[:same_address_as_id] == 'false' &&
                   !pii_from_user.has_key?(:address1)
-        redirect_to idv_in_person_ssn_url
+        return if request.referer == idv_in_person_verify_info_url
+        redirect_to idv_in_person_verify_info_url unless !user_session[:idv][:ssn]
+      end
+
+      def confirm_ssn_step_needed
+        if pii_from_user&.has_key?(:address1) && !user_session[:idv][:ssn]
+          redirect_to idv_in_person_ssn_url
+        end
```

Comment:
I think if you make the suggested change to address step above, you won't need this before_action.